Session Issue #3

kobito-kun opened the pull request:
Issue #3

kobito-kun committed:
Session Issue #4

amazingproducer commented on quackfactory/cogs/session.py:
```diff
@@ -0,0 +1,34 @@
+    channel_id = 756574459556659323
+    message = f"Hey! {member} needs a rubber duck!"
+
+    if before.channel is None and after.channel.id == channel_id:
```
Is there a reason you chose to only react to users who were not in other voice channels?

amazingproducer commented on quackfactory/cogs/session.py:
```diff
@@ -0,0 +1,34 @@
+        await after.channel.edit(user_limit=2)
+
+        for member in guild.members:
+            for role in member.roles:
```
Our specification dictates that we notify *available* ducks; please consider using presence intents to retrieve [user status](https://discordpy.readthedocs.io/en/latest/api.html#discord.Status).

amazingproducer reviewed (commented):
This PR appears to solve #3, though it claims to solve #4. I've included some additional comments. Let's see if the rest of the team has any input.

hum commented on quackfactory/cogs/session.py:
```diff
@@ -0,0 +1,34 @@
+intents = discord.Intents.default()
+intents.reactions = True
+intents.members = True
+client = discord.Client(intents=intents)
```
Why are you creating a separate instance of a bot inside of the cog? I recommend to go through the project and follow the code flow. Declaring intents and creating a new bot instance isn't what cogs are for -- these things are already created inside of the application.

PS: You are importing the logging package, yet you aren't using it.

kobito-kun replied:
i'll remove those 👍

kobito-kun commented:
I'm so sorry :-;

kobito-kun committed:
opens a session

fixed so that it would only message the online rubber ducks. Removed unique instance of a bot and 'before.channel == none' which solves the issue of a user changing from 1 vc to another.

kobito-kun commented:
Hello, solved all the issues above. Please re-check. I apologise for not understanding well.

kobito-kun committed:
Update session.py

kobito-kun committed:
Updated session.py

Issues fixed includes such as: removing the new instance of another bot, removing the 'before.channel...' so that users from existing voice chats won't be ignored, fixed the issue where it would message all the rubber ducks; now it will message the "available, online" rubberducks.

amazingproducer commented:
Looks better! Let's see what @hum has to say.

hum commented on quackfactory/cogs/session.py:
```diff
@@ -0,0 +1,28 @@
+        for member in guild.members:
+            for role in member.roles:
+                if role.name == "Test": # Rubberduck role idk why but it doesnt work using a variable 
+                    if str(member.status) == "online" or member.status == discord.Status.online:
```
`if str(member.status) == "online" or member.status == discord.Status.online:`
These both evaluate the same thing.
```suggestion
                    if member.status == discord.Status.online:
```

hum replied:
You also need to add the proper intents in config.py for the status to work.

kobito-kun replied:
Hi, I know this but I'll remove it.

hum commented on quackfactory/cogs/session.py:
```diff
@@ -0,0 +1,28 @@
+    def __init__(self, bot):
+        self.bot = bot
+
+@client.event
```
```suggestion
@commands.Cog.listener()
```
You are using the wrong bot object for this to work as a cog.

it has to be the `discord.commands` bot instance. Please refer to [reactions.py](https://github.com/silobusters/quackfactory/blob/master/quackfactory/cogs/reactions.py)

kobito-kun replied:
Alright.

kobito-kun committed:
minor fixes

kobito-kun committed:
Update config.py

kobito-kun committed:
Update config.py

kobito-kun commented:
Alright, added presences intents, used correct bot object and cleaned check member status.

amazingproducer commented on quackfactory/cogs/session.py:
```diff
@@ -21,7 +22,7 @@ async def on_voice_state_update(member, before, after):
         for member in guild.members:
             for role in member.roles:
                 if role.name == "Test": # Rubberduck role idk why but it doesnt work using a variable 
```
Perhaps you could compare the [role id](https://discordpy.readthedocs.io/en/latest/api.html?highlight=role%20id#discord.Role.id) to the named variable for the role.

kobito-kun replied:
Alright.

hum commented on quackfactory/cogs/session.py:
```diff
@@ -0,0 +1,34 @@
+        self.bot = bot
+
+@client.event
+async def on_voice_state_update(member, before, after):
```
Due to the nature of Python relying on indentation to recognise which methods belong to which class, you need to indent this to be inside of the session class.

kobito-kun committed:
creates a session

now uses role id to identify role and indented.

hum commented:
LGTM
